Observer/OfflinePaymentObserver.php

```diff
@@ -108,4 +115,18 @@ public function execute(Observer $observer)
 
         return $this;
     }
+
+    /**
+     * calculates expiry time for paynow charge
+     * @return string
+     */
+    private function getPaynowChargeExpiryTime() {
+        $limit = date("d-m-Y")." 22:59:59";
+        $now = date("d-m-Y H:i:s");
+        if(strtotime($now) > strtotime($limit)) {
+            return date("d-m-Y" , strtotime("+1 day"))." 23:59:59";
+        } else {
+            return date("d-m-Y H:i:s" , strtotime(date("Y-m-d 23:59:59")));
+        }
+    }
```

Comment:
I hate time in PHP... but this is also fun to play with... could we try this one?
```php
$timestamp = time() > strtotime('tomorrow - 1seconds') ? strtotime('tomorrow + 1day - 1second') : strtotime('tomorrow - 1seconds');
return date("d-m-Y H:i:s" , $timestamp);
```

Reply:
This looks way easier, using this now. Thanks